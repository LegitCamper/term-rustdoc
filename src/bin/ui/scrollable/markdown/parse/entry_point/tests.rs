use super::{markdown_iter, parse};
use insta::{assert_debug_snapshot as snap, assert_display_snapshot as shot};

#[test]
fn parse_markdown() {
    let doc = r#"
# h1 `code`

aaa b *c* d **e**. ~xxx~ z.

1 *c **ss** d sadsad xxx* `yyyy`

```
let a = 1;
```

> rrr sss 
> tt

- [x] done!
    - nested list
- [ ] undone
    1. *a*
    2. `b`
"#;
    snap!(markdown_iter(doc).collect::<Vec<_>>());
    shot!(parse(doc), @r###"
    # h1 `code`

    aaa b c d e. xxx z.

    1 c ss d sadsad xxx `yyyy`

    let a = 1;

    rrr sss tt

    * [x] done!
      * nested list
    * [ ] undone
      1. a
      2. `b`

    "###);
}

#[test]
fn parse_markdown_links() {
    let doc = "
[a](b), [c], [e][c].

[c]: d

[`f`][c].

## h2 [c] [`h`][c]
";
    snap!(markdown_iter(doc).collect::<Vec<_>>());
    shot!(parse(doc), @r###"
    a, c, e.

    `f`.

    ## h2 c `h`

    "###);
}
