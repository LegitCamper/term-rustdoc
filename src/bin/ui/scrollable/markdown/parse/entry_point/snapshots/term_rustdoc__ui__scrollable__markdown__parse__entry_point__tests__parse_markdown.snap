---
source: src/bin/ui/scrollable/markdown/parse/entry_point/tests.rs
expression: "markdown_iter(doc).collect::<Vec<_>>()"
---
[
    (
        Start(
            Heading {
                level: H1,
                id: None,
                classes: [],
                attrs: [],
            },
        ),
        1..13,
    ),
    (
        Text(
            Borrowed(
                "h1 ",
            ),
        ),
        3..6,
    ),
    (
        Code(
            Borrowed(
                "code",
            ),
        ),
        6..12,
    ),
    (
        End(
            Heading(
                H1,
            ),
        ),
        1..13,
    ),
    (
        Start(
            Paragraph,
        ),
        14..42,
    ),
    (
        Text(
            Borrowed(
                "aaa b ",
            ),
        ),
        14..20,
    ),
    (
        Start(
            Emphasis,
        ),
        20..23,
    ),
    (
        Text(
            Borrowed(
                "c",
            ),
        ),
        21..22,
    ),
    (
        End(
            Emphasis,
        ),
        20..23,
    ),
    (
        Text(
            Borrowed(
                " d ",
            ),
        ),
        23..26,
    ),
    (
        Start(
            Strong,
        ),
        26..31,
    ),
    (
        Text(
            Borrowed(
                "e",
            ),
        ),
        28..29,
    ),
    (
        End(
            Strong,
        ),
        26..31,
    ),
    (
        Text(
            Borrowed(
                ". ",
            ),
        ),
        31..33,
    ),
    (
        Start(
            Strikethrough,
        ),
        33..38,
    ),
    (
        Text(
            Borrowed(
                "xxx",
            ),
        ),
        34..37,
    ),
    (
        End(
            Strikethrough,
        ),
        33..38,
    ),
    (
        Text(
            Borrowed(
                " z.",
            ),
        ),
        38..41,
    ),
    (
        End(
            Paragraph,
        ),
        14..42,
    ),
    (
        Start(
            Paragraph,
        ),
        43..76,
    ),
    (
        Text(
            Borrowed(
                "1 ",
            ),
        ),
        43..45,
    ),
    (
        Start(
            Emphasis,
        ),
        45..68,
    ),
    (
        Text(
            Borrowed(
                "c ",
            ),
        ),
        46..48,
    ),
    (
        Start(
            Strong,
        ),
        48..54,
    ),
    (
        Text(
            Borrowed(
                "ss",
            ),
        ),
        50..52,
    ),
    (
        End(
            Strong,
        ),
        48..54,
    ),
    (
        Text(
            Borrowed(
                " d sadsad xxx",
            ),
        ),
        54..67,
    ),
    (
        End(
            Emphasis,
        ),
        45..68,
    ),
    (
        Text(
            Borrowed(
                " ",
            ),
        ),
        68..69,
    ),
    (
        Code(
            Borrowed(
                "yyyy",
            ),
        ),
        69..75,
    ),
    (
        End(
            Paragraph,
        ),
        43..76,
    ),
    (
        Start(
            CodeBlock(
                Fenced(
                    Borrowed(
                        "",
                    ),
                ),
            ),
        ),
        77..95,
    ),
    (
        Text(
            Borrowed(
                "let a = 1;\n",
            ),
        ),
        81..92,
    ),
    (
        End(
            CodeBlock,
        ),
        77..95,
    ),
    (
        Start(
            BlockQuote,
        ),
        97..113,
    ),
    (
        Start(
            Paragraph,
        ),
        99..113,
    ),
    (
        Text(
            Borrowed(
                "rrr sss",
            ),
        ),
        99..106,
    ),
    (
        SoftBreak,
        107..108,
    ),
    (
        Text(
            Borrowed(
                "tt",
            ),
        ),
        110..112,
    ),
    (
        End(
            Paragraph,
        ),
        99..113,
    ),
    (
        End(
            BlockQuote,
        ),
        97..113,
    ),
    (
        Start(
            List(
                None,
            ),
        ),
        114..179,
    ),
    (
        Start(
            Item,
        ),
        114..144,
    ),
    (
        TaskListMarker(
            true,
        ),
        116..119,
    ),
    (
        Text(
            Borrowed(
                "done!",
            ),
        ),
        120..125,
    ),
    (
        Start(
            List(
                None,
            ),
        ),
        128..144,
    ),
    (
        Start(
            Item,
        ),
        128..144,
    ),
    (
        Text(
            Borrowed(
                "nested list",
            ),
        ),
        132..143,
    ),
    (
        End(
            Item,
        ),
        128..144,
    ),
    (
        End(
            List(
                false,
            ),
        ),
        128..144,
    ),
    (
        End(
            Item,
        ),
        114..144,
    ),
    (
        Start(
            Item,
        ),
        144..179,
    ),
    (
        TaskListMarker(
            false,
        ),
        146..149,
    ),
    (
        Text(
            Borrowed(
                "undone",
            ),
        ),
        150..156,
    ),
    (
        Start(
            List(
                Some(
                    1,
                ),
            ),
        ),
        159..179,
    ),
    (
        Start(
            Item,
        ),
        159..168,
    ),
    (
        Start(
            Emphasis,
        ),
        164..167,
    ),
    (
        Text(
            Borrowed(
                "a",
            ),
        ),
        165..166,
    ),
    (
        End(
            Emphasis,
        ),
        164..167,
    ),
    (
        End(
            Item,
        ),
        159..168,
    ),
    (
        Start(
            Item,
        ),
        170..179,
    ),
    (
        Code(
            Borrowed(
                "b",
            ),
        ),
        175..178,
    ),
    (
        End(
            Item,
        ),
        170..179,
    ),
    (
        End(
            List(
                true,
            ),
        ),
        159..179,
    ),
    (
        End(
            Item,
        ),
        144..179,
    ),
    (
        End(
            List(
                false,
            ),
        ),
        114..179,
    ),
]
